packages/desktop-client/src/components/reports/spreadsheets/filterHiddenItems.tsx

```diff
@@ -0,0 +1,13 @@
+function filterHiddenItems(item, data) {
+  let filtered = data.filter(asset =>
+    item.uncat_id
+      ? (item.transfer ? asset.transferAccount : !asset.transferAccount) &&
+        (item.category ? true : !asset.category) &&
+        (item.offBudget ? asset.accountOffBudget : !asset.accountOffBudget)
+      : true,
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏this expression is very difficult to read. Is there anything we could do to improve the readability of it? Perhaps breaking it down into smaller conditional statements that return early?

i.e.
```
if (something) {
  return true;
}

if (somethingElse) {
   return false;
}

return true;
```

Reply:
Made a code suggestion here: https://github.com/actualbudget/actual/pull/1988#discussion_r1414326875